lib/puppet/type/logical_volume.rb

```diff
@@ -13,9 +13,19 @@
         raise ArgumentError, "Volume names must be entirely unqualified"
       end
     end
+
+    # This is a bit of a hack, drags the volume_group up from the provider if
+    # it was specified, allowing `puppet resource` and otrher things that rely
+    # on an indirector search to work
+    munge do |name|
+      if @resource.original_parameters[:provider].respond_to? :volume_group
+        @resource[:volume_group] = @resource.original_parameters[:provider].volume_group
+      end
+      name
+    end
   end
 
-  newparam(:volume_group) do
+  newproperty(:volume_group) do
```

Comment:
Changing this, will cause issues. It may be backwards incompatible

Reply:
The problem with this being a param is that it's not possible to pull a param from `puppet resource` which will then cause all the tasks to fail. By definition this should have always been a property as it is able to be read back from the system and isn't just something that is provided on creation. Can we run the beaker suite or something to verify what issues it might create as this change is critical to this entire PR (except the facts)

Reply:
@tphoney Any feedback on this?